### PR TITLE
@craigspaeth => Add better logging and update superagent

### PR DESCRIPTION
**Pull request by kanaabe**
The repro steps are [here](https://github.com/artsy/positron/issues/873). I can only reproduce it on staging which means we'll need to bubble up better error messages. I can determine it's caused from the client, and farthest traceable error in papertrail is: `superagent timeout 10000ms GET https://stagingwriter.artsy.net/api/articles`

Also updates `superagent` to 3.x.x from 1.x.x. I did some spot checking to make nothing breaks locally.

### CODE REVIEW

File: api/apps/articles/routes.coffee [hunk 1]
```diff
@@ -16,6 +16,7 @@ User = require '../users/model.coffee'
       'Pass published=true to only view published articles.')
 
   Article.where req.query, (err, results) ->
+    debug err if err
```
Comment (craigspaeth) [inline api/apps/articles/routes.coffee]: I would think [this `debug` in the error handler middleware](https://github.com/artsy/positron/blob/master/api/lib/middleware.coffee#L25) would take care of that no?

**Comment by craigspaeth** [comment]
👍  to updating superagent, but see my comment above about the use of debug. Otherwise looks good.

File: api/apps/articles/routes.coffee (hunk 1 above)
Comment (kanaabe) [inline api/apps/articles/routes.coffee]: Hmm yeah you're right that it should get handed in the error handler. So assuming then that any error from the API fetch/mongo stuff is handled, what else could cause the hanging `superagent` request? `superagent timeout 10000ms GET https://stagingwriter.artsy.net/api/articles`. It's always after this error (called from the client) that memory consumption gets piled up.

**Comment by kanaabe** [comment]
Ok I'll merge this without the debug. Basically just another shrinkwrap and update to `superagent`

**Pull request merged**